week-1-ts-ecosystem/w01e04-light-me-up/src/models/LightBulb.ts

```diff
@@ -0,0 +1,47 @@
+import { PowerSource } from './PowerSource';
+
+const DEFAULT_POWER_CONSUMPTION = 20;
+const INTERVAL_DURATION = 1000;
+
+class LightBulb {
+  #powerConsumption = DEFAULT_POWER_CONSUMPTION;
+  #powerSource: PowerSource;
+
+  public constructor(powerSource: PowerSource) {
+    this.#powerSource = powerSource;
+  }
+
+  public flash(): void {
+    this.#powerSource.consume(this.#powerConsumption);
+  }
+
+  public switchOn(numberOfSeconds: number): Promise<void> {
+    console.log('Light bulb switched on');
+
+    return new Promise((resolve, reject) => {
```

Comment:
Tutaj super - dokładnie o to chodziło w kontekście obsługi błędu wewnątrz `callback`